File: src/langs/rust/wheel-part.rs
extern crate rand;

use rand::Rng;

#[no_mangle]
pub fn name() -> *const u8 {
  let n = b"Rust\0";
  return n as *const u8;
}

#[allow(non_snake_case)]
#[no_mangle]
pub fn feelingLucky() -> f64 {
  let mut rng = rand::thread_rng();
  let y: f64 = rng.gen_range(1.0..101.0);
  return y.floor() + 1.0
}

File: src/langs/zig/wheel-part.zig
const std = @import("std");
const Rng = std.rand.Pcg;

extern fn seed() callconv(.C) f64;

var rng: Rng = undefined;

pub fn main() void {
    rng = Rng.init(@intFromFloat(seed()));
}

export fn name() callconv(.C) [*c]const u8 {
    return "Zig";
}

export fn feelingLucky() callconv(.C) f64 {
    return @floatFromInt(rng.random().intRangeAtMost(u7, 1, 100));
}
